src/hooks/useDomainAvailability.ts

```diff
@@ -0,0 +1,64 @@
+import { useQuery } from 'react-query';
+import { useBuyDomain } from './useBuyDomain';
+import { useZnsSdk } from './useZnsSdk';
+import { MIN_DOMAIN_NAME_LENGTH } from '../constants/domains';
+
+export interface UseDomainAvailabilityReturn {
+	isLoading: boolean;
+	isDomainAvailable: boolean;
+	isDomainSearchEnabled: boolean;
+	domainPrice?: string;
+}
+
+export const useDomainAvailability = (): UseDomainAvailabilityReturn => {
+	const sdk = useZnsSdk();
+	const { domainName } = useBuyDomain();
+
+	const isDomainSearchEnabled = domainName.length >= MIN_DOMAIN_NAME_LENGTH;
+
+	// Query
+	const { isLoading, data: { isDomainAvailable, domainPrice } = {} } = useQuery(
+		`buy-domains-search-${domainName}`,
```

Comment:
[Query keys should be arrays](https://tanstack.com/query/v4/docs/guides/query-keys). This one applies for the other queries in this PR but just putting it on this one!

Reply:
Updated